app/packages/looker/src/worker/threed-label-processor.ts

```diff
@@ -158,8 +158,8 @@ const PainterFactory3D = (
       ymax
     );
 
-    const tlx = (xScale * (x - lx / 2 + (xmax - xmin) / 2)) / canvasWidth; // top left x, normalized to [0, 1]
-    const tly = (yScale * (-y - ly / 2 + (ymax - ymin) / 2)) / canvasHeight; // top left y, normalized to [0, 1]
+    const tlx = (xScale * (x - lx / 2 - xmin)) / canvasWidth; // top left x, normalized to [0, 1]
+    const tly = (yScale * (-y - ly / 2 + ymax)) / canvasHeight; // top left y, normalized to [0, 1]
```

Comment:
`(xmax - xmin) / 2` recalibrates the origin, but it needs to be offsetted by `xmax - xmin`. `(xmas - xmin - (xmax - xmin)) / 2 = xmin`. Same applies in `tly` calculation.